crates/ironrdp-pdu/src/rdp/headers.rs

```diff
@@ -322,6 +322,18 @@ pub enum ShareDataPdu {
     ShutdownDenied,
     SuppressOutput(SuppressOutputPdu),
     RefreshRectangle(RefreshRectanglePdu),
+    Update(Vec<u8>),
```

Comment:
I mostly agree! Here is an alternative:

```rust
struct ShareDataPduRaw<'a> {
    kind: ShareDataPduType,
    data: Cow<'a, [u8]>,
}

impl ShareDataPdu<'_> {
     fn parse(&self) -> Result<ShareDataPdu, …> { … }
}
```

With `ShareDataPdu` only including supported PDUs, and allowing graceful handling both known and unknown PDU types.

That being said, if the plan is to support all the different types soon enough, it’s probably fine to just add the new variants now instead. I don’t think the protocol will change that much, and if it does, it will likely be controlled by compatibly flags during the handshake.

Reply:
I'm going to punt on this one for now.